ast35/Python/ast.c

```diff
@@ -1304,10 +1320,16 @@ handle_keywordonly_args(struct compiling *c, const node *n, int start,
                 if (!arg)
                     goto error;
                 asdl_seq_SET(kwonlyargs, j++, arg);
-                i += 2; /* the name and the comma */
+                i += 1 + (TYPE(CHILD(n, i + 1)) == COMMA); /* the name and the comma, if present */
```

Comment:
Or maybe the comma part should even just be another case of the switch -- that's more straightforward and I believe is equivalent given that we got a valid CST in the first place. Might be slightly slower, though, as the CPU would lose the chance to branch-predict that there is in fact a comma there.